pkg/share/share.go

```diff
@@ -118,9 +117,7 @@ func MatchesFilter(share *collaboration.Share, filter *collaboration.Filter) boo
 	case collaboration.Filter_TYPE_GRANTEE_TYPE:
 		return share.Grantee.Type == filter.GetGranteeType()
 	case collaboration.Filter_TYPE_EXCLUDE_DENIALS:
-		// This filter type is used to filter out "denial shares". These are currently implemented by having the permission "0".
-		// I.e. if the permission is 0 we don't want to show it.
-		return int(conversions.RoleFromResourcePermissions(share.Permissions.Permissions).OCSPermissions()) != 0
+		return share.Permissions.Permissions.DenyGrant
```

Comment:
You should use permissionsNone here because you want to filter out the denials not the DenyGrant